core/scheduler/service/job_run_service.go

```diff
@@ -780,6 +788,24 @@ func (s *JobRunService) raiseJobRunStateChangeEvent(jobRun *scheduler.JobRun) {
 	s.eventHandler.HandleEvent(schedulerEvent)
 }
 
+func (s *JobRunService) registerSLA(ctx context.Context, operatorRunID string, operatorStartTime time.Time,
+	operatorType scheduler.OperatorType, jobName, operatorName string, slaAlertConfigs []*scheduler.SLAAlertConfig,
+) error {
+	me := errors.NewMultiError("errorInRegisterSLA")
+	for _, slaAlertConfig := range slaAlertConfigs {
+		slaBoundary := operatorStartTime.Add(slaAlertConfig.DurationThreshold)
+		slaDescription := fmt.Sprintf("%s: %s, durationSLA: %s", operatorType, operatorName, slaAlertConfig.DurationThreshold.String())
+		err := s.slaRepo.RegisterSLA(ctx, jobName, operatorName, operatorType.String(), operatorRunID, slaBoundary, slaDescription)
+		if err != nil {
+			//	also emmit critical metric here, such that Optimus Devs can be alerted
+			errMsg := fmt.Sprintf("error registering sla for operator Run Id: %s, Type: %s, Sla: %s, err: %s", operatorRunID, operatorType.String(), slaDescription, err.Error())
+			me.Append(errors.Wrap(scheduler.EntityEvent, errMsg, err))
+			s.l.Error(errMsg)
+		}
+	}
+	return me.ToErr()
+}
+
 func (s *JobRunService) createOperatorRun(ctx context.Context, event *scheduler.Event, operatorType scheduler.OperatorType) error {
```

Comment:
we could have kept create operator run & register sla's independent we don't need to keep them in a same function.